lib/ParseSIL/ParseSIL.cpp

```diff
@@ -885,6 +887,119 @@ void SILParser::convertRequirements(SILFunction *F,
   }
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+/// Parse a SIL constant value (one of the following categories):
+/// - An integer datatype and literal (i32 42).
+/// - A floating point datatype and literal (f64 3.14).
+/// - A metatype (the instance type is parsed) ($Float).
+/// - An aggregate ([1, 2, 3]).
+/// Returns true on error.
+static bool parseSymbolicValue(SymbolicValue &value, SILParser &SP,
+                               SILBuilder &B) {
+  auto &P = SP.P;
+  auto &allocator = B.getModule().getASTContext().getAllocator();
+
+  if (P.Tok.is(tok::identifier)) {
+    // Bitwidth of the integer/floating point literal.
+    unsigned width;
+    // Handle integer literals.
+    if (P.Tok.getText().startswith("i")) {
+      // Parse integer datatype.
+      P.consumeStartingCharacterOfCurrentToken();
+      if (SP.parseInteger(width, diag::sil_const_expected_int_datatype))
+        return true;
+
+      // Parse integer value.
+      bool negative = false;
+      if (P.Tok.isAnyOperator() && P.Tok.getText() == "-") {
+        negative = true;
+        P.consumeToken();
+      }
+      APInt intValue(width, 0);
+      if (SP.parseInteger(intValue, diag::sil_graph_op_expected_attr_value))
+        return true;
+
+      // Negate and truncate value, if necessary.
+      if (negative)
+        intValue = -intValue;
+      if (intValue.getBitWidth() != width)
+        intValue = intValue.zextOrTrunc(width);
+
+      value = SymbolicValue::getInteger(intValue, allocator);
+      return false;
+    }
+    // Handle floating point literals.
+    if (P.Tok.getText().startswith("f")) {
+      // Parse floating point datatype.
+      auto datatypeToken = P.Tok;
+      P.consumeStartingCharacterOfCurrentToken();
+      if (SP.parseInteger(width, diag::sil_const_expected_fp_datatype))
+        return true;
+
+      // Parse floating point value.
+      double floatValue;
+      P.Tok.getText().getAsDouble(floatValue);
```

Comment:
Is this what the float_literal instruction does?  You should follow its implementation for the parsing logic.

Reply:
The `float_literal` instruction actually parses hex floating point values and prints the decimal representation in a comment:
`%5 = float_literal $Builtin.FPIEEE80, 0x4000C8F5C28F5C28F5C3 // 3.1400000000000000001`

Would you prefer that I use hexadecimal as well?
I couldn't think of a nice way to print decimal representations in a comment, so I just used decimal instead.

Reply:
After discussion, we decided that:
* Parser should accept both decimal/hexadecimal formats.
* Printer should print hexadecimal only.